[internal/otelarrow] Flaky test disabled: TestIntegrationMemoryLimited

### Component(s)

internal/otelarrow

### Describe the issue you're reporting

Hit on #34358 see https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10203882356/job/28231140032?pr=34358#step:6:518

```terminal
=== FAIL: test TestIntegrationSelfTracing (11.03s)
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:369: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:369
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:205
        	            				C:/hostedtoolcache/windows/go/1.21.12/x64/src/runtime/asm_amd64.s:1650
        	Error:      	Received unexpected error:
        	            	rpc error: code = Canceled desc = send wait: context deadline exceeded
        	Test:       	TestIntegrationSelfTracing
    e2e_test.go:272: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:272
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:418
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:220
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:476
        	Error:      	Not equal: 
        	            	expected: 10000
        	            	actual  : 4664
        	Test:       	TestIntegrationSelfTracing
```

This is still happening in Windows CIs though with different messages https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11370646308/job/31630914777
```
=== Failed
=== FAIL: test TestIntegrationMemoryLimited (37.46s)
make[2]: *** [../../Makefile.Common:131: test] Error 1
    e2e_test.go:100: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:100
make[1]: *** [Makefile:200: internal/otelarrow] Error 2
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:892
make[1]: *** Waiting for unfinished jobs....
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:702
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:682
        	            				C:/hostedtoolcache/windows/go/1.22.8/x64/src/runtime/asm_amd64.s:1695
        	Error:      	"4.166666666s" is not less than "4.1333912s"
        	Test:       	TestIntegrationMemoryLimited

=== FAIL: test TestIntegrationMemoryLimited (re-run 1) (44.79s)
    e2e_test.go:100: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/otelarrow/test/e2e_test.go:100
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:892
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:702
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/arrow/arrow.go:682
        	            				C:/hostedtoolcache/windows/go/1.22.8/x64/src/runtime/asm_amd64.s:1695
        	Error:      	"4.166666666s" is not less than "3.9117612s"
        	Test:       	TestIntegrationMemoryLimited

DONE 2 runs, 50 tests, 2 failures in 178.112s
✓  . (1.129s)
```

--- a/internal/otelarrow/test/e2e_test.go
+++ b/internal/otelarrow/test/e2e_test.go
@@ -93,13 +93,6 @@ func (tc *testConsumer) ConsumeTraces(ctx context.Context, td ptrace.Traces) err
 	timeout := time.Until(dead)
 
 	require.Equal(tc.t, tc.expectDeadline, hasDeadline, "deadline set or not set: %v", timeout)
-	if tc.expectDeadline {
-		// expect allows 1/6 of the deadline to elapse in transit,
-		// so 1m becomes 50s.
-		expect := tc.expCfg.TimeoutSettings.Timeout * 5 / 6
-		require.Less(tc.t, expect, timeout)
-		require.Greater(tc.t, tc.expCfg.TimeoutSettings.Timeout, timeout)
-	}
 
 	return tc.sink.ConsumeTraces(ctx, td)
 }

